src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  let newState = state;
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = Object.assign(newState, action.extraData);
```

Comment:
You don't need to use Object.assign, instead use the spread operator to merge the extraData into the state object.